content/custom-decorators.md

```diff
@@ -137,6 +137,55 @@ async findOne(firstName) {
 
 You can use this same decorator with different keys to access different properties. If the `user` object is deep or complex, this can make for easier and more readable request handler implementations.
 
+#### Type checking data on Custom Decorators
+
+(Typescript only)
+
+Let's take another look at the `user.decorator` file:
+
+```typescript
+@@filename(user.decorator)
+import { createParamDecorator } from '@nestjs/common';
+
+export const User = createParamDecorator((data: string, req) => {
+  return data ? req.user && req.user[data] : req.user;
+});
+```
+
+As we have set the `data` parameter to be of type 'string' only, that implies
+
+1. Inside the Decorator function the `data` type (in this case, string) will be enforced and checked as it would on any other typescript function parameter;
+2. When implementing and using the decorator on a `Controller` the string data type will also be enforced when passing the params;
+
+This means that :
+
+```typescript
+@@filename()
+@Get()
+async findOne(@User('firstName') firstName: string) { // THIS WORKS!
+  console.log(`Hello ${firstName}`);
+}
+@Get()
+async wrongFindOne(@User(200) firstName: string) { // THIS FAILS! As the User decorator expects a string
+  console.log(`Hello ${firstName}`);
+}
+```
+
+If we had not defined a type, the `data` parameter would default to `any`;
+
+In practice, typing allows us to build safer decorators that can be checked while coding and on build, avoiding unexpected behavior both when implementing on `Controllers` as on coding its own functionality;
+
+The `createParamDecorator` also allows the definition of type through Generics, so the `user.decorator` can be rewritten as:
```

Comment:
To be honest, I'd just add this line (and the code below) after [Line#111](https://github.com/nestjs/docs.nestjs.com/pull/881/files#diff-c815b0ab017a4ee9e5d1e4170c61a3e2R111).

The rest of it kind of belongs in a blog/article somewhere, not a website documentation. 😄